src/NHibernate/Hql/Ast/ANTLR/HqlSqlWalker.cs

```diff
@@ -725,6 +723,7 @@ void PrepareFilterParameter()
 
 			// Generate an explicit join for the root dot node.   The implied joins will be collected and passed up
 			// to the root dot node.
+			dot.SkipSemiResolve = true;
```

Comment:
It's either implicit or entity join. So skip any processing that's not fully resolve this node.